systemtest/agentconfig_test.go

```diff
@@ -103,6 +105,14 @@ func TestAgentConfig(t *testing.T) {
 		assert.Equal(t, configured2, settings)
 		assert.NotEqual(t, etags[url], resp.Header.Get("Etag"))
 	}
+
+	t.Log("waiting for the agent_config metrics to be published...")
+	waitForLogMessage(t, srv, "bulk request completed: 2 indexed", 1)
```

Comment:
ExpectMinDocs will retry for up to 10 seconds. Is it needed/useful to wait for the metrics to be published first?

Reply:
Looking closer, I see that we hard code the agentcfg.Reporter to report every 30 seconds. Would it be reasonable to make that configurable (does not need to be documented or advertised), and reduce the interval for tests?

Reply:
I thought about making it configurable, but it seemed a little too much for a bug fix. I can make flush period configurable in this PR if you feel that's the best approach.

On the other hand, we're building the agent image to inject an APM server binary into it which will take 15-20s. The flush timer is running in the `apm-server` binary that we run in line 49 and the image building is accounted into that, but making it configurable would most likely reduce the test runtime by almost half.

Reply:
> I thought about making it configurable, but it seemed a little too much for a bug fix.

Probably a bit too much for backporting to 7.16.

How about we extract the systemtest changes into a followup, making the config change at the same time? That doesn't need to be backported to 7.16 -- particularly since these metrics an't used yet.

> On the other hand, we're building the agent image to inject an APM server binary into it which will take 15-20s.

That time is (should be?) paid only once for all the system tests -- not for every test.